pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -115,6 +117,70 @@ func (configgen *ConfigGeneratorImpl) BuildListeners(node *model.Proxy,
 	return builder.getListeners()
 }
 
+func (configgen *ConfigGeneratorImpl) BuildListenerTLSContext(serverTLSSettings *networking.ServerTLSSettings, proxy *model.Proxy, transportProtocol istionetworking.TransportProtocol) *auth.DownstreamTlsContext {
+	alpnByTransport := util.ALPNHttp
+	if transportProtocol == istionetworking.TransportProtocolQUIC {
+		alpnByTransport = util.ALPNHttp3OverQUIC
```

Comment:
I am not sure we can support QUIC, we don't have redirection support

Reply:
@howardjohn Actually I just yanked this code from gateway and created a new func so that both GW and Sidecar can call it. So this may make sense for GW.